### PR TITLE
Chunk Tile List

### PR DESCRIPTION
**Pull request by ThatDan123**
-Splits the tile list in 20x20 chunks

-Remove lists if theres no items in them

### CODE REVIEW

File: UnityProject/Assets/Scripts/Tilemaps/Utils/TileList.cs [hunk 1]
```diff
@@ -1,156 +1,191 @@
...
 		}
-		var list = new List<RegisterTile>();
-		foreach (var x in objectsOut)
+
+		public void InvokeOnObjects(IRegisterTileAction action, Vector3Int localPosition)
 		{
-			if (x.ObjectType == type) {
-				list.Add(x);
+			tempRegisterTiles.Clear();
+			tempRegisterTiles.AddRange(Get(localPosition));
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Tilemaps/Utils/TileList.cs:126]: How thread safe  is this tempRegisterTiles?

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Tilemaps/Utils/TileList.cs:126]: probably isnt safe

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tilemaps/Utils/TileList.cs:126]: might be worth just slapping a Lock around it?

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Tilemaps/Utils/TileList.cs:126]: I havent changed any logic, just moved it about. Was working fine before

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tilemaps/Utils/TileList.cs:126]: ok cool

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**